### PR TITLE
Agent header support

### PR DESCRIPTION
**Pull request by sacOO7**
- Added agent header support with proper lib versioning.
- Updated README for release process

### CODE REVIEW

**Comment by sacOO7** [comment]
@qsdstefan can you check if we can add a test to check for `agent-header` being sent. 
You can check for the one here ->
https://github.com/ably/ably-php-laravel/blob/e3f564a2f526e1348f34a3d3b71f5dfc17bb0854/tests/AblyLaravelTest.php#L137

**Comment by sacOO7** [comment]
Implemented as a part of https://github.com/ably/ably-common/pull/170

**Review by qsdstefan — APPROVED** [review]
Looks correct!

File: README.md [hunk 1]
```diff
@@ -6,14 +6,15 @@
...
 
-This implements ably broadcaster as a independent service provider library for [Laravel](https://laravel.com/) using [ably-php](https://github.com/ably/ably-php). This library works with [ably-js](https://github.com/ably/ably-js) based [ably-laravel-echo](https://github.com/ably-forks/echo) framework having enhanced features at client side. Main aim is to replace old [pusher-client based ably broadcaster](https://laravel.com/docs/9.x/broadcasting#client-ably).
+This implements ably broadcaster as a independent service provider library for [Laravel](https://laravel.com/) using [ably-php](https://github.com/ably/ably-php). This library works with [ably-js](https://github.com/ably/ably-js) based [ably-laravel-echo](https://github.com/ably-forks/echo) client framework with enhanced features. Main aim is to replace old [pusher-client based ably broadcaster](https://laravel.com/docs/9.x/broadcasting#client-ably).
```
Comment (owenpearson) [inline README.md:9]:
```suggestion
This implements ably broadcaster as a independent service provider library for [Laravel](https://laravel.com/) using [ably-php](https://github.com/ably/ably-php). This library works with the [ably-js](https://github.com/ably/ably-js) based [ably-laravel-echo](https://github.com/ably-forks/echo) client framework with enhanced features. This project is the successor to the [pusher-client based ably broadcaster](https://laravel.com/docs/9.x/broadcasting#client-ably).
```

File: tests/AblyBroadcasterTest.php [hunk 2]
```diff
@@ -325,4 +327,34 @@ protected function getMockRequestWithoutUserForChannel($channel, $token)
...
+        $this->lastUrl = $url;
+        $this->lastHeaders = $headers;
+
+        // mock response to /time
+        return array(
+            'headers' => "HTTP/1.1 200 OK\n",
+            'body' => array(round(microtime(true) * 1000 ), 0),
+        );
+    }
+}
```
Comment (owenpearson) [inline tests/AblyBroadcasterTest.php:360]: missing newline character at end of file

File: tests/AblyBroadcasterTest.php [hunk 3]
```diff
@@ -325,4 +327,34 @@ protected function getMockRequestWithoutUserForChannel($channel, $token)
...
+    public $lastHeaders;
+
+    public function request($method, $url, $headers = array(), $params = array())
+    {
+        $this->lastUrl = $url;
+        $this->lastHeaders = $headers;
+
+        // mock response to /time
+        return array(
+            'headers' => "HTTP/1.1 200 OK\n",
```
Comment (owenpearson) [inline tests/AblyBroadcasterTest.php:356]: Clearly this test is passing but I'm a bit confused by this line, the field is called 'headers' but it contains an HTTP start-line? And is there not a neater way to do this than to write raw HTTP response text?

**Review by owenpearson — CHANGES_REQUESTED** [review]
Looks good on the whole, a few things that need addressing.

Also, it seems like this PR conflates the agent header support with some unrelated documentation changes, I would prefer to have separate PRs for these if possible.

File: tests/AblyBroadcasterTest.php (hunk 3 above)
Comment (sacOO7) [reply tests/AblyBroadcasterTest.php:356]: I think this is how we have done it in `ably-php` and `ably-php-laravel`. This is just a copy pasted mock from there ... maybe this is the only way to do it.

File: tests/AblyBroadcasterTest.php (hunk 2 above)
Comment (sacOO7) [reply tests/AblyBroadcasterTest.php]: updated!

File: tests/AblyBroadcasterTest.php (hunk 3 above)
Comment (owenpearson) [reply tests/AblyBroadcasterTest.php:356]: OK that's fair, I don't particularly like it but I won't block this PR on it 🙂

**Review by owenpearson — APPROVED** [review]

**Comment by sacOO7** [comment]
Merging the PR since it's approved : )

**Pull request merged**